src/main/scala/zio/redis/options/zrangebylex.scala

```diff
@@ -0,0 +1,22 @@
+package zio.redis.options
+
+object zrangebylex {
+
+  sealed trait Minimum
```

Comment:
The docs for `ZLEXCOUNT` states `The min and max arguments have the same meaning as described for `ZRANGEBYLEX`. So, instead of re-defining it for every command, I'd just leave this one, and use it as an input for other commands referring to it.

**Edit:** I realized I might have mislead you in our chat regarding flattening of the options package. Sorry for that. Let's not postpone that and flatten the options package immediately:

```scala
object options {
// all custom options
}
```

You can call the file `options.scala` and put it in the same package with `Input`. That would help reducing the noise, e.g. no need for limit inputs specific to each command.

Reply:
As discussed in our chat the plan is to group the types for the commands in a similar fashion like it is done on the [Redis commands documentation page itself ](https://redis.io/commands).
Additional there will be a shared module to declare cross-cutting command options.